prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -1468,7 +1468,7 @@ removeTrailingWhitespace = go (RecordedWhitespace [] 0)
     commitSpaces [] 0 = id
     commitSpaces [] 1 = SChar ' '
     commitSpaces [] n = SText n (T.replicate n " ")
-    commitSpaces [i] n = SLine i . commitSpaces [] n
+    commitSpaces [_] n = SLine 0 . commitSpaces [] n
```

Comment:
Yeah, this is wrong:

```
> removeTrailingWhitespace (SLine 2 (SChar 'x' SEmpty))
SLine 0 (SChar 'x' SEmpty)
```

This should still be 

```
SLine 2 (SChar 'x' SEmpty)
```

I'm a bit disappointed that there are no tests that caught it though! :/